elastic/kibana
# [Response Ops][Actions] Improve error handling in ES Index action response

Resolves https://github.com/elastic/kibana/issues/162989

## Summary

Updates error parsing for bulk index action in ES index connector. Previously, we were parsing for the `index.error` key within the returned bulk response but it looks like since switching to `op_type: create`, the response is actually returned inside the `create` key. This PR updates the error handling to key off of the `errors` flag in the bulk response and then parse out the error from within the items array, regardless of of the key. Bulk API with sample responses can be found here: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#bulk-api-response-body.

🙏 Check for any edge cases I may have missed while parsing the error response

## To Verify

Create an ES index connector that writes to an existing mapped index (I loaded the `kibana_sample_data_logs` dataset). Test the connector and try to index a field with an incompatible mapping (example, index a string into a `long` field) and verify that the failure to index is correct returned as an action execution error.
